src/windows.rs

```diff
@@ -15,15 +15,42 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
+
+#![allow(unsafe_code)]
+
 extern crate winreg;
+extern crate winapi;
 
+use self::winapi::windef::HWND;
+use self::winapi::winnt::LPCWSTR;
 use self::winreg::RegKey;
 use self::winreg::enums::HKEY_CURRENT_USER;
 use app::App;
 
 use errors::*;
+use std::ffi::OsStr;
+use std::os::windows::ffi::OsStrExt;
 use std::path::Path;
-use std::process::Command;
+use std::ptr;
+
+fn to_wide_chars(s: &str) -> Vec<u16> {
+    OsStr::new(s)
+        .encode_wide()
+        .chain(Some(0).into_iter())
+        .collect::<Vec<_>>()
+}
+
+#[link(name = "shell32")]
+extern "system" {
```

Comment:
I guess the convention is to explicitly state the ABI - so `extern "C"` here perhaps (unless that gives a compile error for some reason)